rest/src/main/java/com/netflix/conductor/rest/rbac/RbacAccessAspect.java

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright 2024 Netflix, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.netflix.conductor.rest.rbac;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Pointcut;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.HttpClientErrorException;
+
+import com.netflix.conductor.common.metadata.BaseDef;
+import com.netflix.conductor.common.metadata.workflow.StartWorkflowRequest;
+import com.netflix.conductor.common.model.BulkResponse;
+import com.netflix.conductor.core.exception.NotFoundException;
+
+@Aspect
+@Component
+public class RbacAccessAspect {
+
+    private final RbacDbHandler handler;
+
+    private final RbacHttpFilter filter;
+
+    private Object[] arguments;
+
+    public RbacAccessAspect(RbacDbHandler handler, RbacHttpFilter filter) {
+        this.handler = handler;
+        this.filter = filter;
+    }
+
+    @Pointcut("@annotation(com.netflix.conductor.rest.rbac.annotations.RbacAdminAccess)")
+    public void adminAccessMethods() {}
+
+    @Pointcut("@annotation(com.netflix.conductor.rest.rbac.annotations.RbacBulkAccess)")
+    public void rbacBulkAccess() {}
+
+    @Pointcut("@annotation(com.netflix.conductor.rest.rbac.annotations.RbacPathVarObject)")
+    public void rbacPathVarObject() {}
+
+    /**
+     * Executes methods with admin access control.
+     *
+     * <p>Proceeds only in situations when user is admin. Otherwise, throws 401.
+     *
+     * @param joinPoint The proceeding join point, enabling interception of the method call.
+     * @return The result of the intercepted method call.
+     * @throws Throwable Throws any throwable exception that occurs during method execution.
+     */
+    @Around("adminAccessMethods()")
```

Comment:
Why you dont set @Pointcut directly to @Around() annotation without method creating?

Reply:
Directly set.